pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -1049,16 +1040,19 @@ func TestResolvePipelineRun_PipelineTaskHasNoResources(t *testing.T) {
 
 	getTask := func(name string) (v1beta1.TaskInterface, error) { return task, nil }
 	getTaskRun := func(name string) (*v1beta1.TaskRun, error) { return &trs[0], nil }
-	getClusterTask := func(name string) (v1beta1.TaskInterface, error) { return clustertask, nil }
 	getCondition := func(name string) (*v1alpha1.Condition, error) { return nil, nil }
 	pr := v1beta1.PipelineRun{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "pipelinerun",
 		},
 	}
-	pipelineState, err := ResolvePipelineRun(context.Background(), pr, getTask, getTaskRun, getClusterTask, getCondition, pts, providedResources)
-	if err != nil {
-		t.Fatalf("Did not expect error when resolving PipelineRun without Resources: %v", err)
+	pipelineState := PipelineRunState{}
+	for _, task := range pts {
+		ps, err := ResolvePipelineRunTask(context.Background(), pr, getTask, getTaskRun, getCondition, task, providedResources)
+		if err != nil {
+			t.Fatalf("Error getting tasks for fake pipeline %s: %s", p.ObjectMeta.Name, err)
```

Comment:
We may want to use `t.Errorf` here (to gather all the tasks error on a test failure)